### ISSUE TITLE
Add explanation when test and finetune option in links.BatchNormalization.__call__ is both True

### ISSUE THREAD

**Issue by delta2323** [chainer/chainer#804]
[Description](https://github.com/pfnet/chainer/blob/master/chainer/links/normalization/batch_normalization.py#L76) of `BatchNormalization.__call__()` implies that `BatchNormalization` can take both of testing and finetuning mode simultaneously when `test` option and `finetune` option is both `True`.

But [document](https://github.com/pfnet/chainer/blob/master/chainer/links/normalization/batch_normalization.py#L15) in `links.BatchNormalization` implicitly assume that training, finetuning and testing modes are disjoint, meaning that `BatchNormalization` takes either one of three modes (correct me if it is not the writer's intent).

**Comment by delta2323** [comment]
My understanding is that `BatchNormalization` has two modes simultaneously: one is either training/testing mode and the other is either finetuning/not finetuning mode. If it is correct it is better that fix the explanation about three modes.
